projects/ngx-charts-on-fhir/src/lib/fhir-mappers/simple-medication-mapper.service.ts

```diff
@@ -4,72 +4,53 @@ import { MedicationOrder } from 'fhir/r2';
 import { merge } from 'lodash-es';
 import { DataLayer } from '../data-layer/data-layer';
 import { Mapper } from '../fhir-converter/multi-mapper.service';
-import { ChartAnnotation } from '../utils';
-import { TIME_SCALE_OPTIONS, LINEAR_SCALE_OPTIONS, ANNOTATION_OPTIONS } from './fhir-mapper-options';
+import { TIME_SCALE_OPTIONS, CATEGORY_SCALE_OPTIONS } from './fhir-mapper-options';
 import { FhirMappersModule } from './fhir-mappers.module';
 
-/** Required properties for mapping an Observation with [SimpleMedicationMapper] */
+/** Required properties for mapping a MedicationOrder with [SimpleMedicationMapper] */
 export type SimpleMedication = {
   medicationCodeableConcept: {
     text: string;
   };
   dateWritten: string;
 } & MedicationOrder;
 export function isMedication(resource: MedicationOrder): resource is SimpleMedication {
-  return !!(
-    resource.resourceType === 'MedicationOrder' &&
-    resource.prescriber?.reference &&
-    resource.dosageInstruction &&
-    resource.medicationCodeableConcept
-  );
+  return !!(resource.resourceType === 'MedicationOrder' && resource.dateWritten && resource.medicationCodeableConcept?.text);
 }
```

Comment:
This function is a type guard for the parameter type that is passed into the `SimpleMedicationMapper#map` function. It's important that the logic here correctly discriminates the `SimpleMedication` type so that we don't get runtime type errors for resources with missing properties (and also to keep the mapping code simple). For example, if the `SimpleMedication` type asserts that `dateWritten` will always be present, then the `map` function can use it without sprinkling null checks everywhere, but this means the type guard (`canMap` function) needs to return false if `dateWritten` is undefined.

I also removed the extra properties that were not used by the mapper. If properties are not relevant, there's no need to check them in the type guard (unless of course you **want** to filter out those resources).